config/routes.rb

```diff
@@ -1,31 +1,82 @@
 Triannon::Engine.routes.draw do
 
-  resources :annotations, except: [:update, :edit],
-    # show action must explicitly forbid "new", "iiif" and "oa" as id values;  couldn't
-    #   figure out how to do it with regexp constraint since beginning and end regex
-    #   matchers aren't allowed when enforcing formats for segment (e.g. :id)
-    constraints: lambda { |request|
-                            id = request.env["action_dispatch.request.path_parameters"][:id]
-                            id !~ /^iiif$/ && id !~ /^oa$/ && id !~ /^search$/
-                 } do
-    collection do
-      get 'search', to: 'search#find'
-    end
-  end
+  # 1. can't use resourceful routing because of :anno_root (dynamic path segment)
```

Comment:
Do you have routing specs for this? I can't tell because this pull request has so many changes.